server/gitrest/packages/gitrest-base/src/test/summaries.spec.ts

```diff
@@ -154,9 +155,18 @@ const getFsManagerFactory = (
 				keyExpireAfterSeconds: 60 * 60, // 1 hour
 			},
 		});
+		const redisOptions: RedisOptions = {
+			host: "localhost",
+			port: 6379,
+			connectTimeout: 10000,
+			maxRetriesPerRequest: 20,
+			enableAutoPipelining: false,
+			enableOfflineQueue: true,
+		};
+		const testRedisClientConnectionManager = new TestRedisClientConnectionManager(redisOptions);
```

Comment:
It seems like tests are passing, so I'm probably not understanding how `ioredis-mock` works, but the `TestRedisClientConnectionManager` class creates a new mock client every time `getRedisClient()` is called. For 1, this is significantly different from how the normal RedisClientConnectionManager is setup, so maybe it would be good to align those. More importantly though, I thought each RedisMock client would have its own "Redis" datastore, so creating a new one for each call would be creating a blank datastore for each call, which would break all the tests...

Reply:
Ah, this probably works because ioredis-mock v6 made the "datastore" (context) shared

> Before v6, each instance of ioredis-mock lived in isolation
> ...
> In v6 the [internals were rewritten](https://github.com/stipsan/ioredis-mock/pull/1110) to behave more like real life redis, if the host and port is the same, the context is now shared:
> — https://www.npmjs.com/package/ioredis-mock#from-v5-to-v6

Reply:
I wonder if we should still change `TestRedisClientConnectionManager` in a follow-up PR back in R11s to be setup more like `RedisClientConnectionManager`, where the client is created within the `constructor` on initialization, and `getRedisClient()` is a simple `getter`.

Reply:
Yeah I can raise a follow up PR to change that behaviour in the testing class.